assets/src/models/createDetourMachine.ts

```diff
@@ -1,19 +1,68 @@
-import { setup, ActorLogicFrom, InputFrom } from "xstate"
-import { Route, RoutePattern } from "../schedule"
+import { setup, assign, fromPromise, ActorLogicFrom, InputFrom } from "xstate"
+import { Route, RouteId, RoutePattern } from "../schedule"
+import { fetchRoutePatterns } from "../api"
 
 export const createDetourMachine = setup({
   types: {} as {
     context: {
-      route: Route
-      routePattern: RoutePattern
-    }
+      route?: Route
+      routePattern?: RoutePattern
 
-    input: {
-      // Caller has target route pattern
-      route: Route
-      routePattern: RoutePattern
+      routePatterns?: RoutePattern[]
     }
-    events: { type: "detour.edit.done" } | { type: "detour.edit.resume" }
+
+    input:
+      | {
+          // Caller has target route pattern
+          route: Route
+          routePattern: RoutePattern
+        }
+      | {
+          // Caller has target route
+          route: Route
+          routePattern: undefined
+        }
+      | {
+          // Caller has no prior selection
+          route: undefined
+          routePattern: undefined
+        }
```

Comment:
```suggestion
```

I don't think this is a use case that we currently want to support. The two entrypoints are that we already have a route pattern (from the search map), or that we have a route but no route pattern ( from the route ladders).

Reply:
This is not something we _currently_ have implemented, but it would not be difficult to update the `<DummyDetourPage>` to use this behavior.

I think we _do_ want this behavior, as per this [figma slide: Detour Opened from Detours Page](https://www.figma.com/design/XGmmwVROXdOvnWtJeUFC50/Skate-%7C-Unplanned-Detours-%7C-Hand-Off?node-id=2855-6968&t=XF9NKdbU2WB3iojO-1), but you could argue it shouldn't be implemented yet because that's not what we're doing yet. 

IMO, I'd leave it so that it's clear in the future what the machine _can_ handle, but we could drop it.

Reply:
Eh that's fair. Since we're pretty certain we'll want to support it, I'm okay leaving it in.